caches.go

```diff
@@ -92,6 +93,33 @@ func (rcc *clientRegionCache) clientDown(c hrpc.RegionClient) map[hrpc.RegionInf
 	return downregions
 }
 
+// Collects information about the clientRegion cache and stores them in two maps to reduce
+// duplication of data. We do this in one function to avoid running the iterations twice
+func (rcc *clientRegionCache) ClientRegionCacheDebugInfo(
+	keyRegionInfosMap map[string]hrpc.RegionInfo,
+	clientRegionsMap map[string]hrpc.RegionClient) map[string][]string {
+	// key = RegionClient memory address , value = List of RegionInfo addresses
+	clientRegionCacheMap := map[string][]string{}
+
+	rcc.m.RLock()
+	for client, reginfos := range rcc.regions {
+		clientRegionInfoMap := make([]string, 0)
```

Comment:
Pass len(reginfos) as the second arg to "make". You could then do the following on line 112:

  clientRegionInfoMap[i] = fmt.Sprintf(..)

instead of using "append"